frameworks/spring-boot/src/main/java/io/ap4k/spring/SpringPropertiesHolder.java

```diff
@@ -0,0 +1,23 @@
+package io.ap4k.spring;
+
+import io.ap4k.WithProject;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+public interface SpringPropertiesHolder extends WithProject {
+
+    AtomicReference<Map<String, Object>> springProperties = new AtomicReference<>(null);
+
+    default Map<String, Object> getSpringProperties() {
+        if (springProperties.get() == null) {
+            final Map<String, Object> properties = new HashMap<>();
+            properties.putAll(getProject().parseAppResourceFile("application.properties"));
```

Comment:
I am wondering if we should also take profiles into consideration.

Reply:
Sorry, I just got notified about this and hadn't seen it.
We could do that yes, *if* the profile is defined in the file and not by some other means (by in my experience that doesn't happen often).